asmrun/Makefile

```diff
@@ -142,20 +142,26 @@ install:
 $(LINKEDFILES): %.c: ../byterun/%.c
 	$(LN) $< $@
 
-%.d.$(O): %.c
-	$(CC) -c $(DFLAGS) $(CPPFLAGS) $(OUTPUTOBJ)$@ $<
+# The COMPILE_C_FILE function below receives as argument the pattern
+# that corresponds to the name of the generated object file
+define COMPILE_C_FILE
+$(1): %.c
+	$$(CC) -c $$(OC_CFLAGS) $$(OC_CPPFLAGS) $$(OUTPUTOBJ)$$@ $$<
+endef
 
-%.i.$(O): %.c
-	$(CC) -c $(IFLAGS) $(CPPFLAGS) $(OUTPUTOBJ)$@ $<
+object_types := %.$(O) %.d.$(O) %.i.$(O) %.p.$(O) %.pic.$(O)
 
-%.p.$(O): %.c
-	$(CC) -c $(PFLAGS) $(CPPFLAGS) $(OUTPUTOBJ)$@ $<
+$(foreach object_type, $(object_types), \
+  $(eval $(call COMPILE_C_FILE,$(object_type))))
```

Comment:
I know it's equivalent, but part of me would prefer the definition of the build rule (via `eval`) to be physically below the setting up of dependencies (in this case the env. variables)

Reply:
I agree with @dra27: for readability it should be moved down.